### PR TITLE
Add com.github.calo001.luna

### PR DESCRIPTION
**Pull request by calo001**
A calendar Widget with awesome colors.

### CODE REVIEW

File: com.github.calo001.luna.json [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    "app-id" : "com.github.calo001.luna",
+    "base" : "io.elementary.BaseApp",
+    "base-version" : "juno",
+    "runtime" : "org.gnome.Platform",
+    "runtime-version" : "3.30",
+    "sdk" : "org.gnome.Sdk",
+    "command" : "com.github.calo001.luna",
+    "finish-args" : [
+        "--share=ipc", "--socket=x11",
+        "--socket=wayland",
```
Comment (bilelmoussaoui) [inline com.github.calo001.luna.json]: Does it really work with Wayland?

File: com.github.calo001.luna.json [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        "--filesystem=xdg-run/dconf", "--filesystem=~/.config/dconf:ro",
+        "--talk-name=ca.desrt.dconf", "--env=DCONF_USER_CONFIG_DIR=.config/dconf"
+    ],
+    "build-options" : {
```
Comment (bilelmoussaoui) [inline com.github.calo001.luna.json]: Those are default now. You can remove them. They are used from freedesktop runtime 18.08 which is the base of the gnome & kde runtimes.

File: com.github.calo001.luna.json [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        "*.a"
+    ],
+    "modules" : [
+        {
+            "name" : "Luna",
+            "buildsystem" : "meson",
+            "config-opts" : [
+                "--libdir=lib"
+            ],
+            "builddir" : true,
```
Comment (bilelmoussaoui) [inline com.github.calo001.luna.json]: Not needed.

File: com.github.calo001.luna.json [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        "/share/pkgconfig",
+        "/share/vala",
+        "*.la",
+        "*.a"
+    ],
+    "modules" : [
+        {
+            "name" : "Luna",
+            "buildsystem" : "meson",
+            "config-opts" : [
```
Comment (bilelmoussaoui) [inline com.github.calo001.luna.json]: Not needed. it's already the default value of libdir.

File: com.github.calo001.luna.json [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+                "--libdir=lib"
+            ],
+            "builddir" : true,
+            "sources" : [
+                {
+                    "type" : "git",
+                    "url" : "https://github.com/calo001/luna.git",
+                    "commit" : "f973efd6acf523fe194925096c3ef570e42781b0"
```
Comment (bilelmoussaoui) [inline com.github.calo001.luna.json]: Using release tarballs is faster than cloning the repository. You can keep this until you have a new release with whatever fixes.

File: com.github.calo001.luna.json [hunk 6]
```diff
@@ -0,0 +1,50 @@
+{
+    "app-id" : "com.github.calo001.luna",
+    "base" : "io.elementary.BaseApp",
+    "base-version" : "juno",
+    "runtime" : "org.gnome.Platform",
+    "runtime-version" : "3.30",
+    "sdk" : "org.gnome.Sdk",
```
Comment (bilelmoussaoui) [inline com.github.calo001.luna.json]: I don't see any specific reasons why you're using GNOME Sdk here. You should use FreeDesktop one here. It already has Gtk there. There are not any specific GNOME libs that your app is using

File: com.github.calo001.luna.json (hunk 1 above)
Comment (calo001) [reply com.github.calo001.luna.json]: I'm not sure, I was reviewing some similar projects, I see that they include support for wayland, would it be wrong to add it?

File: com.github.calo001.luna.json (hunk 2 above)
Comment (calo001) [reply com.github.calo001.luna.json]: Fixed :+1:

File: com.github.calo001.luna.json (hunk 3 above)
Comment (calo001) [reply com.github.calo001.luna.json]: Fixed :+1:

File: com.github.calo001.luna.json (hunk 4 above)
Comment (calo001) [reply com.github.calo001.luna.json]: Fixed :+1:

File: com.github.calo001.luna.json (hunk 6 above)
Comment (calo001) [reply com.github.calo001.luna.json]: Fixed :+1:

**Comment by nedrichards** [comment]
bot, build com.github.calo001.luna
testbot, build com.github.calo001.luna

**Comment by scx** [comment]
There is no `launchable` tag in the AppData file.
https://github.com/calo001/luna/blob/f973efd6acf523fe194925096c3ef570e42781b0/data/com.github.calo001.luna.appdata.xml.in

See also:
 https://github.com/flathub/flathub/wiki/AppData-Guidelines#launchable

File: com.github.calo001.luna.json (hunk 1 above)
Comment (bilelmoussaoui) [reply com.github.calo001.luna.json]: Well, the application itself puts a window in a specific position which is not possible with Wayland.

File: com.github.calo001.luna.json (hunk 2 above)
Comment (bilelmoussaoui) [reply com.github.calo001.luna.json]: You have also removed dconf settings permissions :/

**Comment by nedrichards** [comment]
bot, build com.github.calo001.luna
testbot, build com.github.calo001.luna

**Comment by nedrichards** [comment]
bot, build com.github.calo001.luna

**Comment by nedrichards** [comment]
@calo001 Does this only work properly on Elementary? (not a problem, just checking) when I launch it I get the word `Time!` printed on the terminal a lot and a nice icon but no window ;-)

**Comment by bilelmoussaoui** [comment]
@nedrichards Are you using Wayland? If yes, then this app is not compatible with it at all

**Comment by bilelmoussaoui** [comment]
testbot, build com.github.calo001.luna

**Comment by scx** [comment]
bot, build com.github.calo001.luna

**Comment by nedrichards** [comment]
I'm using Endless OS but if others have tested and it works fine that's good enough for me.

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/com.github.calo001.luna

**Pull request closed without merging**